@gmarull looks like you need to update mcuboot also.

```
 /__w/zephyr/bootloader/mcuboot/boot/zephyr/main.c: In function 'main':
/__w/zephyr/bootloader/mcuboot/boot/zephyr/main.c:449:13: error: Macro is deprecated [-Werror]
  449 |     if (!flash_device_get_binding(DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL)) {
      |             ^~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/bootloader/mcuboot/boot/zephyr/main.c:451:13: error: Macro is deprecated [-Werror]
  451 |        DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL);
      |             ^~~~~~~~~~~~~~~~~~~~~
```

--- a/drivers/flash/flash_shell.c
+++ b/drivers/flash/flash_shell.c
@@ -6,6 +6,7 @@
  */
 
 #include <zephyr.h>
+#include <devicetree.h>
 
 #include <shell/shell.h>
 #include <sys/util.h>
@@ -18,11 +19,8 @@
 #define BUF_ARRAY_CNT 16
 #define TEST_ARR_SIZE 0x1000
 
-#ifdef DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
-#define FLASH_DEV_NAME DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
-#else
-#define FLASH_DEV_NAME ""
-#endif
+static const struct device *zephyr_flash_controller =
+	DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_flash_controller));
 
 static uint8_t __aligned(4) test_arr[TEST_ARR_SIZE];
 
@@ -33,12 +31,27 @@ static int parse_helper(const struct shell *shell, size_t *argc,
 	char *endptr;
 
 	*addr = strtoul((*argv)[1], &endptr, 16);
-	*flash_dev = device_get_binding((*endptr != '\0') ? (*argv)[1] :
-			FLASH_DEV_NAME);
-	if (!*flash_dev) {
-		shell_error(shell, "Flash driver was not found!");
+
+	if (*endptr != '\0') {
+		/* flash controller from user input */
+		*flash_dev = device_get_binding((*argv)[1]);
+		if (!*flash_dev) {
+			shell_error(shell, "Given flash device was not found");
+			return -ENODEV;
+		}
+	} else if (zephyr_flash_controller != NULL) {
+		/* default to zephyr,flash-controller */
+		if (!device_is_ready(zephyr_flash_controller)) {
+			shell_error(shell, "Default flash driver not ready");
+			return -ENODEV;
+		}
+		*flash_dev = zephyr_flash_controller;
+	} else {
+		/* no flash controller given, no default available */
+		shell_error(shell, "No flash device specified (required)");
 		return -ENODEV;
 	}
+
 	if (*endptr == '\0') {
 		return 0;
 	}

--- a/include/devicetree/zephyr.h
+++ b/include/devicetree/zephyr.h
@@ -46,6 +46,10 @@
 /**
  * @def DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
  *
+ * @deprecated Use @c DT_LABEL(DT_CHOSEN(zephyr_flash_controller)) instead. If
+ * used to to obtain a device instance with device_get_binding(), consider using
+ * @c DEVICE_DT_GET(DT_CHOSEN(zephyr_flash_controller)).
+ *
  * @brief If there is a chosen node zephyr,flash-controller property which has
  *        a label property, that property's value. Undefined otherwise.
  */
@@ -60,7 +64,7 @@
 
 #if DT_NODE_HAS_PROP(DT_CHOSEN(zephyr_flash_controller), label)
 #define DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL \
-	DT_LABEL(DT_CHOSEN(zephyr_flash_controller))
+	__DEPRECATED_MACRO DT_LABEL(DT_CHOSEN(zephyr_flash_controller))
 #endif
 
 /**

--- a/tests/drivers/flash/src/main.c
+++ b/tests/drivers/flash/src/main.c
@@ -12,7 +12,7 @@
 
 #if (CONFIG_NORDIC_QSPI_NOR - 0)
 #define NORDIC_QSPI_NOR_NODE DT_INST(0, nordic_qspi_nor)
-#define FLASH_DEVICE DT_LABEL(NORDIC_QSPI_NOR_NODE)
+#define FLASH_NODEID NORDIC_QSPI_NOR_NODE
 #define FLASH_TEST_REGION_OFFSET 0xff000
 
 #if DT_NODE_HAS_PROP(NORDIC_QSPI_NOR_NODE, size_in_bytes)
@@ -23,18 +23,18 @@
 
 #elif defined(CONFIG_FLASH_MCUX_FLEXSPI_NOR)
 
-#define FLASH_DEVICE DT_LABEL(DT_INST(0, nxp_imx_flexspi_nor))
+#define FLASH_NODEID DT_INST(0, nxp_imx_flexspi_nor)
 #define FLASH_TEST_REGION_OFFSET FLASH_AREA_OFFSET(storage)
 #define TEST_AREA_MAX ((FLASH_AREA_SIZE(storage)) + (FLASH_TEST_REGION_OFFSET))
 #elif defined(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G)
 
-#define FLASH_DEVICE DT_LABEL(DT_INST(0, nxp_imx_flexspi_mx25um51345g))
+#define FLASH_NODEID DT_INST(0, nxp_imx_flexspi_mx25um51345g)
 #define FLASH_TEST_REGION_OFFSET FLASH_AREA_OFFSET(storage)
 #define TEST_AREA_MAX ((FLASH_AREA_SIZE(storage)) + (FLASH_TEST_REGION_OFFSET))
 #else
 
 /* SoC embedded NVM */
-#define FLASH_DEVICE DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
+#define FLASH_NODEID DT_CHOSEN(zephyr_flash_controller)
 
 #ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
 #define FLASH_TEST_REGION_OFFSET FLASH_AREA_OFFSET(image_1_nonsecure)
@@ -50,7 +50,7 @@
 #define EXPECTED_SIZE	256
 #define CANARY		0xff
 
-static const struct device *flash_dev;
+static const struct device *flash_dev = DEVICE_DT_GET(FLASH_NODEID);
 static struct flash_pages_info page_info;
 static uint8_t __aligned(4) expected[EXPECTED_SIZE];
 
@@ -58,7 +58,8 @@ static void test_setup(void)
 {
 	int rc;
 
-	flash_dev = device_get_binding(FLASH_DEVICE);
+	zassert_true(device_is_ready(flash_dev), NULL);
+
 	const struct flash_parameters *flash_params =
 			flash_get_parameters(flash_dev);
 

